src/OpenSage.Game/Network/SkirmishManager.cs

```diff
@@ -13,20 +13,61 @@ namespace OpenSage.Network
 {
     public abstract class SkirmishManager
     {
-        private string _connectionKey = string.Empty; // TODO: maybe use this for password protection?
+        protected static readonly NLog.Logger Logger = NLog.LogManager.GetCurrentClassLogger();
 
         protected Game _game;
+
+        public SkirmishManager(Game game, bool isHosting)
+        {
+            _game = game;
+
+            IsHosting = isHosting;
+            Settings = new SkirmishGameSettings(isHosting);
+        }
+
+        public bool IsHosting { get; protected set; }
+        public SkirmishGameSettings Settings { get; protected set; }
+        public IConnection Connection { get; protected set; }
+
+        public abstract bool IsStartButtonEnabled();
+        public abstract Task HandleStartButtonClickAsync();
+
+        public virtual void Stop()
+        {
+        }
+    }
+
+    public sealed class LocalSkirmishManager : SkirmishManager
```

Comment:
I added a `LocalSkirmishManager` class for non-network skirmish games (next to the existing implementations for Host and Client).